### ISSUE TITLE
[HOLD for payment 2023-09-07] [$1000] Chat comment linking redirects to 'Hmm... it's not here' instead of the intended chat

### ISSUE THREAD

**Issue by perunt** [Expensify/App#23711]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
beta flag: commentLinking
1. Open the chat.
2. Copy link to the message
3. Put this link in the chat
4. Click on this link.

## Expected Result:
The application should redirect to the referenced chat.

## Actual Result:
Instead of navigating to the intended chat, the application returns a 'Hmm... it's not here' page.

## Workaround:
do not need as the comment linking feature is in development 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.46-0
**Reproducible in staging?:**y
**Reproducible in production?:**y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:** @perunt
**Slack conversation:**

https://github.com/Expensify/App/assets/48593211/b855e3bb-f460-4fbe-ad8e-345ac3c8ebb1

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01aef64f5f78bfffbd</li>
        <li>Upwork Job ID: 1685030150680666112</li>
        <li>Last Price Increase: 2023-08-04</li>
    </ul>
</details>

**Comment by perunt** [comment]
Use beta flag 'commentLinking' to enable the possibility of 'copy link to message'

https://github.com/Expensify/App/assets/48593211/dcd3a7da-12f0-4f99-add8-74913b2f1bc7

**Comment by perunt** [comment]
**Proposal**
**Please re-state the problem that we are trying to solve in this issue.**
For further implementation of the comment linking feature, we need this functionality of navigating to chat works. Now we have error page instead of redirection to the proper report room

**What is the root cause of that problem?**
Navigator link formatting

**What changes do you think we should make in order to solve the problem?**
we need update [this](https://github.com/Expensify/App/blob/65ffc1023920635d2a20e76617b1e06a9c871ce7/src/ROUTES.js#L44) part of the code
and put 
```
REPORT_WITH_ID: 'r/:reportID/:reportActionID?',
```
also expand propTypes [here](https://github.com/Expensify/App/blob/d0dc566281a882ab300f328f484bea738121c5e3/src/pages/home/ReportScreen.js#L38-L44) as well
```
/** Navigation route context info provided by react navigation */
route: PropTypes.shape({
    /** Route specific parameters used on this screen */
    params: PropTypes.shape({
        /** The ID of the report this screen should display */
        reportID: PropTypes.string,

        /** The reportActionID to scroll to */
        reportActionID: PropTypes.string,
    }).isRequired,
}).isRequired,
```

**Comment by roryabraham** [comment]
@alexpensify FWIW I asked @perunt to create this issue

**Comment by perunt** [comment]
@dukenv0307, it seems my solution found its way into your edited comment 🤣 
 
 Anyway, in the scope of this task, I would also fix the format of the link created, as now it needs to be revised. Can I do it in this scope, or should I create a new issue?

**Comment by alexpensify** [comment]
@roryabraham thanks for the update. 

I've assigned external. @mananjadhav when you get a chance please review the proposals. Thanks!

**Comment by mananjadhav** [comment]
@roryabraham do we have any preference on the URL? Path params vs query params. I am inclined to use @perunt's proposal as it is inline with rest of our URLs, but do we need any kind of identifier between the reportId and reportActionId?

**Comment by perunt** [comment]
@mananjadhav it's preparing to [this](https://github.com/Expensify/App/issues/20282#issuecomment-1652253817) task.

**Comment by alexpensify** [comment]
@roryabraham when you get a chance, can you review the comments starting here: https://github.com/Expensify/App/issues/23711#issuecomment-1656336329

**Comment by alexpensify** [comment]
@cead22 - Rory is OOO until next week. To keep this moving forward, we need feedback on this question: https://github.com/Expensify/App/issues/23711#issuecomment-1656336329

Thank you!

**Comment by alexpensify** [comment]
Carlos is OOO now, so I'm going to hold for now. I think Rory is back on Monday.

**Comment by alexpensify** [comment]
Update: we need a review here before we can move forward

**Comment by roryabraham** [comment]
Hey, sorry for any confusion here but @perunt was hired separately to work on the comment linking project and wrote a design doc which was reviewed by our team, so this really wasn't meant to be open to external proposals.

Regarding @mananjadhav's question, we're using path params, not query params. The format should be: `/r/:reportID/:reportActionID?`

**Comment by alexpensify** [comment]
Thank you for the update, Rory!

**Comment by perunt** [comment]
@roryabraham, I don't want to distract you, but I also want to create an issue regarding the formatting of the copy link. I recall a conversation about it in the comment linking document, but the issue persists. I'm going to fix it before your PRs are ready.

**Comment by alexpensify** [comment]
@roryabraham and @mananjadhav - since there are various parts to this one, should we put on Hold?

**Comment by alexpensify** [comment]
No update

**Comment by alexpensify** [comment]
Weekly Update: On Hold

**Comment by alexpensify** [comment]
@mananjadhav to prepare for the payment day, can you please complete the checklist? Thanks!

**Comment by alexpensify** [comment]
@mananjadhav - payment day is tomorrow, can yo please complete the checklist? Thanks!

**Comment by mananjadhav** [comment]
@alexpensify thanks for following up. I would treat this as a part of feature request, where we land on the specific action after clicking the link. Here we're only ensuring that it lands to a specific chat instead of `Hmm... it's not here`.

Hence I think this doesn't need a regression test too.

@alexpensify Can you post a payout summary?

**Comment by alexpensify** [comment]
Here is the payment summary: 

-  External issue reporter @perunt $250 
-  Contributor that fixed the issue @perunt $1000 
-  Contributor+ that helped on the issue and/or PR @mananjadhav $1000

Upwork Job: https://www.upwork.com/jobs/~01aef64f5f78bfffbd

*If applicable, the bonuses will be applied on the final payment 

Extra Notes regarding payment: There is no bonus and no penalty, this GH required multiple steps and reviews.

**Comment by alexpensify** [comment]
@mananjadhav - let me know if there are questions or concerns, I will close tomorrow. Thanks!

**Comment by JmillsExpensify** [comment]
$1,000 payment approved via NewDot based on BZ summary.
